truffle/src/main/java/org/jruby/truffle/core/kernel/KernelNodes.java

```diff
@@ -1849,17 +1864,25 @@ private DynamicObject finishFormat(int formatLength, BytesResult result) {
         }
 
         @TruffleBoundary
-        protected CallTarget compileFormat(DynamicObject format, Object[] arguments) {
+        protected CallTarget compileFormat(DynamicObject format, Object[] arguments, boolean isDebug) {
             assert RubyGuards.isRubyString(format);
 
             try {
                 return new PrintfCompiler(getContext(), this)
-                        .compile(Layouts.STRING.getRope(format).getBytes(), arguments);
+                        .compile(Layouts.STRING.getRope(format).getBytes(), arguments, isDebug);
             } catch (InvalidFormatException e) {
                 throw new RaiseException(coreExceptions().argumentError(e.getMessage(), this));
             }
         }
 
+        protected boolean isDebug(VirtualFrame frame) {
+            try {
+                return readDebugGlobalNode.executeBoolean(frame);
```

Comment:
`BooleanCastNode. executeBoolean(frame, value)`.
BTW, we could remove the frame there, it's not needed.